src/store/formulasSlice.ts

```diff
@@ -0,0 +1,156 @@
+import { createSlice, PayloadAction, createSelector } from '@reduxjs/toolkit';
+import { selectLanguage } from './languageSlice';
+import { RootState } from './store';
+import { makeLanguageAndFactories } from '../lib/Factories';
+import { parseConstants, parsePredicates, SyntaxError, parseFormulaStrict } from '@fmfi-uk-1-ain-412/js-fol-parser';
+import { Formula } from '../lib/Formula';
+import { selectValuation } from './valuationSlice';
+
+
+export interface FormulaState {
+  value: string;
+  truthValue: string;
+}
+
+interface FormulasState {
+  formulas: FormulaState[];
+}
+
+const initialState: FormulasState = {
+  formulas: [
+    { value: '', truthValue: "0/1" }
+  ],
+};
+
+const formulasSlice = createSlice({
+  name: 'formulas',
+  initialState,
+  reducers: {
+    addFormulaField(state) {
+      state.formulas.push({ value: '', truthValue: "0/1" });
+    },
+    removeFormulaField(state, action: PayloadAction<number>) {
+      state.formulas.splice(action.payload, 1);
+    },
+    updateFormulaValue(state, action: PayloadAction<{ index: number; value: string }>) {
+      state.formulas[action.payload.index].value = action.payload.value;
+    },
+    updateDropdownValue(state, action: PayloadAction<{ index: number; value: string }>) {
+      state.formulas[action.payload.index].truthValue = action.payload.value;
+    },
+  },
+});
+
+export const { addFormulaField, removeFormulaField, updateFormulaValue, updateDropdownValue } = formulasSlice.actions;
+
+const selectFormula = (state: RootState, index: number) => state.formulas.formulas[index]; 
+const selectFormulas = (state: RootState) => state.formulas.formulas;
+const selectFormulaCount = (state: RootState) => state.formulas.formulas.length;
+
+export interface ParsedFormulaWithValue {
+  parsedFormula?: Formula;
+  truthValue?: boolean | null;
+  error?: SyntaxError;
+};
+
+
+export const selectParsedFormula = createSelector(
+  [selectFormula, selectLanguage, (state: RootState, index: number) => index],
+  (formula, languageState, index): ParsedFormulaWithValue | undefined => {
```

Comment:
Ako moze byt vysledok tohto selectora `undefined`? Nemalo by sa to stat. Odvodi to tak TypeScript?

Zda sa mi, ze ak je tu nejaka ina moznost, tak je to `never`: Funkcia nevrati hodnotu, ale hodi vynimku. Myslim, ze vynimky treba proste `catch`-nut vsetky a vratit ako `error` – bez ohladu na typ. V type ParsedFormulaWithValue by potom mala byt `error?: Error`. Typ vynimky by som zistoval az na mieste pouzitia selectora. Mali by ste mat komponent `Err`, ktory zisti, ci chyba je typu `SyntaxError` a vtedy pouzije komponent `SyntErr` na jej zobrazenie. Inak zobrazi iba `message` chyby (bez vstupneho textu).